src/components/feedback-particle-effect/sure-popup.ts

```diff
@@ -0,0 +1,110 @@
+import { CLICK } from "../../common/event-names";
+import { AudioPlayer } from "../audio-player";
+import YesButton from "../buttons/yes-button";
+import NoButton from "../buttons/no-button";
+
+export default class AreYouSurePopUp {
+  public canvas: HTMLCanvasElement;
+  public context: CanvasRenderingContext2D;
+  public yesButton: YesButton;
+  public noButton: NoButton;
+  public imagesLoaded: boolean = false;
+  public pop_up_image: HTMLImageElement;
+  public yesCallback: Function;
+  public noCallback: Function;
+  audioPlayer: AudioPlayer;
+
+  constructor(canvas, yesCallback, noCallback) {
+    this.canvas = canvas;
+    this.yesCallback = yesCallback;
+    this.noCallback = noCallback;
+
+    const selfIdElement = document.getElementById(
```

Comment:
Can we use canvas passed in constructor to get selfIdElement?

Reply:
yes Rajesh,
i have changed it.